Upgrade Celery and remove some now-unnecessary cruft (bug 1194181)

kmaglione opened the pull request:
We're going to need to add these three packages to the wheelhouse and have Jason change the worker deploy scripts to launch `celery` directly rather than via `manage.py`.

kmaglione commented on apps/amo/celery.py:
```diff
@@ -0,0 +1,33 @@
+@task_failure.connect
+def process_failure_signal(exception, traceback, sender, task_id,
+                           signal, args, kwargs, einfo, **kw):
+    exc_info = (type(exception), exception, traceback)
```
This is a slightly modified version is what was previously imported from `celeryutils`. The rest of that module is no longer required.

magopian commented:
This looks nice, thanks!

Let's wait for PR #673 (which this PR seems based on) to be landed.

magopian commented:
#673 has landed, so now this needs a rebase. I think we discussed what should be done with the "do not merge" commit, but I don't remember, can you please restate it here?

kmaglione commented:
Rebased and removed the 'DO NOT MERGE' commit, but I think we're going to need these new requirement versions in the pyrepo before it passes: https://github.com/mozilla/olympia/pull/678/files#diff-668c57ad76e35851be983271a1a9ac5bR3

kmaglione commented:
Ah. Or not.

kmaglione commented:
Just need @jasonthomas to update the dev servers to launch the Celery workers in the new way, then.

magopian commented:
I'm really curious how the travis build could pass... without the packages updated on pyrepo?!
I've now uploaded all the versions (packages and wheels) to pyrepo.

kmaglione commented:
They were apparently already there.
